components/odh-notebook-controller/controllers/matchers_test.go

```diff
@@ -2,153 +2,477 @@ package controllers
+	someOtherRoute := routev1.Route{
+		Spec: routev1.RouteSpec{
+			Host: "someOtherRouteHost",
+			Path: "someOtherRoutePath",
+		},
+	}
+	shouldPanic := false
+	matcher := BeMatchingK8sResource(someOtherRoute, func(r1 routev1.Route, r2 routev1.Route) bool {
+		if shouldPanic {
+			panic("je nanic")
+		}
+		return r1.Spec.Host == r2.Spec.Host
+	})
+	res, err := matcher.Match(someRoute)
+	assert.NoError(t, err)
+	assert.False(t, res)
+
+	shouldPanic = true
+	msg := matcher.FailureMessage(someRoute)
+	assert.Contains(t, msg, MatcherPanickedMessage)
+	assert.Contains(t, msg, "je nanic")
+
+	assert.Regexp(t, regexp.MustCompile(`-[\pZ\pC]+Host:\s+"someRouteHost",`), msg)
+	assert.Regexp(t, regexp.MustCompile(`\+[\pZ\pC]+Host:\s+"someOtherRouteHost",`), msg)
+
+	assert.Regexp(t, regexp.MustCompile(`-[\pZ\pC]+Path:\s+"someRoutePath",`), msg)
+	assert.Regexp(t, regexp.MustCompile(`\+[\pZ\pC]+Path:\s+"someOtherRoutePath",`), msg)
+}
+
+// Test that map values can be handled correctly without crashing
+// This test specifically targets the "reflect: reflect.Value.Set using unaddressable value" error
+func Test_BeMatchingK8sResource_MapValues(t *testing.T) {
+	someRoute := routev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-route",
+			Labels: map[string]string{
+				"key1": "value1",
+				"key2": "value2",
+			},
+		},
+		Spec: routev1.RouteSpec{
+			Host: "test-host",
+		},
+	}
+	someOtherRoute := routev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-route",
+			Labels: map[string]string{
+				"key1": "different-value1", // Different value
+				"key2": "value2",           // Same value
+				"key3": "value3",           // Extra key
+			},
+		},
+		Spec: routev1.RouteSpec{
+			Host: "test-host",
+		},
+	}
+	// Comparator that only cares about the Host field, not Labels
+	matcher := BeMatchingK8sResource(someOtherRoute, func(r1 routev1.Route, r2 routev1.Route) bool {
+		return r1.Spec.Host == r2.Spec.Host
+	})
+	res, err := matcher.Match(someRoute)
+	assert.NoError(t, err)
+	assert.True(t, res) // Should match because Host is the same
+
+	// Now test with different Host to trigger diff computation
+	someOtherRoute.Spec.Host = "different-host"
+	matcher = BeMatchingK8sResource(someOtherRoute, func(r1 routev1.Route, r2 routev1.Route) bool {
+		return r1.Spec.Host == r2.Spec.Host
 	})
-	res, err := matcher.Match(someHTTPRoute)
+	res, err = matcher.Match(someRoute)
 	assert.NoError(t, err)
 	assert.False(t, res)
 
-	// Basic check that failure message contains expected content
-	msg := matcher.FailureMessage(someHTTPRoute)
-	assert.Contains(t, msg, "somehost.example.com")
-	assert.Contains(t, msg, "otherhost.example.com")
+	// This should not crash when computing the diff, even though Labels differ
+	msg := matcher.FailureMessage(someRoute)
+	assert.NotContains(t, msg, MatcherPanickedMessage)
+	assert.Contains(t, msg, "Minimized diff is")
+	// The minimized diff should not include Label differences since comparator doesn't care about them
```

Comment:
I'm thinking to try Jules or some other agent to try to implement this check. IIUC there is just a comment but the fact is not asserted on.